examples/stm32f1/src/bin/pwm_input.rs

```diff
@@ -0,0 +1,52 @@
+#![no_std]
+#![no_main]
+
+use cortex_m::asm;
+use defmt::*;
+use embassy_executor::Spawner;
+use embassy_stm32::gpio::{Level, Output, Pull, Speed};
+use embassy_stm32::time::khz;
+use embassy_stm32::timer::pwm_input::PwmInput;
+use embassy_stm32::{bind_interrupts, peripherals, timer};
+use embassy_time::Timer;
+use {defmt_rtt as _, panic_probe as _};
+
+/// Connect PB2 and PB10 with a 1k Ohm resistor
+
+#[embassy_executor::task]
+async fn blinky(led: peripherals::PC13) {
+    let mut led = Output::new(led, Level::High, Speed::Low);
+
+    loop {
+        info!("high");
+        led.set_high();
+        Timer::after_millis(300).await;
+
+        info!("low");
+        led.set_low();
+        Timer::after_millis(300).await;
+    }
+}
+
+bind_interrupts!(struct Irqs {
+    TIM2 => timer::CaptureCompareInterruptHandler<peripherals::TIM2>;
+});
+
+#[embassy_executor::main]
+async fn main(spawner: Spawner) {
+    let p = embassy_stm32::init(Default::default());
+    info!("Hello World!");
+
+    unwrap!(spawner.spawn(blinky(p.PC13)));
+
+    let mut pwm_input = PwmInput::new(p.TIM2, p.PA0, Pull::None, khz(10));
+    pwm_input.enable();
+
+    loop {
+        Timer::after_millis(500).await;
+        let _per = pwm_input.get_period_ticks();
```

Comment:
I think we should have async functions to block on *pulse*, fired by interrupts. Like:
```rs
pwm_input.wait_for_pulse().await;
// or
pwm_input.wait_for_capture().await;
// or
pwm_input.wait_for_period().await;
// and
pwm_input.wait_for_pulse_width().await;
```
In order to be able to retrieve the pulse width of exactly the next pulse (my ultrasonic sensor use case)